gnomad/utils/annotations.py

```diff
@@ -1615,64 +1379,73 @@ def gks_compute_seqloc_digest_batch(
     return ht.drop("vrs_json").join(ht_with_location_parsed, how="left")
 
 
-def add_gks_vrs(ht: hl.Table):
+def add_gks_vrs_py(struct_input: hl.struct):
     """
-    Add GKS VRS variant annotation to a hail table.
+    Returns a dictionary containing VRS information from a given Hail Struct
 
-    Annotates ht with GA4GH GKS VRS structure, except for the variant.location._id,
-    which must be computed outside Hail. Use gks_compute_seqloc_digest for this.
+    Dict will have GA4GH GKS VRS structure, except for the variant.location._id,
+    which must be computed outside. Use gks_compute_seqloc_digest for this.
 
-    ht_out.vrs: Struct of the VRS representation of the variant
-    ht_out.vrs_json: JSON string representation of the .vrs struct.
+    :param struct_input: Struct input as the result of running .collect() on a HT
+    :return: Python Dictionary with fields vrs (Struct of the VRS representation of the variant)
     """
-    build_in = get_reference_genome(ht.locus).name
-    chr_in = ht.locus.contig
 
-    vrs_chrom_ids_expr = hl.literal(VRS_CHROM_IDS)
-    chrom_dict = vrs_chrom_ids_expr[build_in]
-    vrs_id = ht.info.vrs.VRS_Allele_IDs[1]
+    build_in = struct_input.locus.reference_genome.name
+    chr_in = struct_input.locus.contig
+
+    chrom_dict = VRS_CHROM_IDS[build_in]
+    vrs_id = struct_input.info.vrs.VRS_Allele_IDs[1]
     vrs_chrom_id = chrom_dict[chr_in]
-    vrs_start_value = ht.info.vrs.VRS_Starts[1]
-    vrs_end_value = ht.info.vrs.VRS_Ends[1]
-    vrs_state_sequence = ht.info.vrs.VRS_States[1]
-
-    ht_out = ht.annotate(
-        vrs=hl.struct(
-            _id=vrs_id,
-            type="Allele",
-            location=hl.struct(
-                _id="",
-                type="SequenceLocation",
-                interval=hl.struct(start=vrs_start_value, end=vrs_end_value),
-                sequence_id=vrs_chrom_id,
-            ),
-            state=hl.struct(
-                type="LiteralSequenceExpression", sequence=vrs_state_sequence
-            ),
-        )
+    vrs_start_value = struct_input.info.vrs.VRS_Starts[1]
+    vrs_end_value = struct_input.info.vrs.VRS_Ends[1]
+    vrs_state_sequence = struct_input.info.vrs.VRS_States[1]
+
+    vrs_dict_out = {
+        "original_struct": struct_input,  # to remove later on
```

Comment:
only pause was that the information from this would be needed for va_py() , but if I'm revising it to only return the vrs information then it's definitely possible to remove here